### ISSUE TITLE
[doc] perlsub "Constant Functions" section out of date

### ISSUE THREAD

**Issue by haarg** [Perl/perl5#19664]
**Where**
perlsub - Constant Functions

**Description**
The "Constant Functions" documentation includes several examples and descriptions that are out of date.

This code is documented as working but issuing a deprecation warning:
```perl
BEGIN {
    my $x = 79907;
    *RT_79908 = sub () { $x };
    $x++;
}
print RT_79908(); # prints 79907
```
This was true in perl 5.22, but it will now fail to compile.

The next example includes:
```perl
# Warns.  Future Perl versions will stop inlining it.
BEGIN {
    my $x;
    $x = 54321;
    *ALSO_INLINED = sub () { $x };
}
```
This also fails to compile.

**Comment by iabyn** [comment]
PR  #19674 contains my proposed fix to the docs.

**Comment by iabyn** [comment]
fixed by f8ed0a9151; closing.
